test/E2ETest/TyePurgeTests.cs

```diff
@@ -0,0 +1,159 @@
+            var tempDirectory = TempDirectory.Create();
+            DirectoryCopy.Copy(projectDirectory.FullName, tempDirectory.DirectoryPath);
+
+            var projectFile = new FileInfo(Path.Combine(tempDirectory.DirectoryPath, "tye.yaml"));
+            var tyeDir = new DirectoryInfo(Path.Combine(tempDirectory.DirectoryPath, ".tye"));
+            using var host = new TyeHost(ConfigFactory.FromFile(projectFile).ToHostingApplication(), Array.Empty<string>())
+            {
+                Sink = sink,
+            };
+
+            await host.StartAsync();
+            try
+            {
+                await WaitUntilSpawned(host.Application);
+
+                var pids = GetAllPids(host.Application);
+
+                Assert.True(Directory.Exists(tyeDir.FullName));
+                Assert.True(AllRunning(pids));
+
+                await host.PurgeAsync();
+                await Task.Delay(500);
+
+                Assert.False(AnyRunning(pids));
+            }
+            finally
+            {
+                await host.StopAsync();
+            }
+
+            try
+            {
+                tempDirectory.Dispose();
+            }
+            catch (UnauthorizedAccessException) { }
+        }
+
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task MultiProjectPurgeTest()
+        {
+            var projectDirectory = new DirectoryInfo(Path.Combine(TestHelpers.GetSolutionRootDirectory("tye"), "samples", "multi-project"));
+            var tempDirectory = TempDirectory.Create();
+            DirectoryCopy.Copy(projectDirectory.FullName, tempDirectory.DirectoryPath);
+
+            var projectFile = new FileInfo(Path.Combine(tempDirectory.DirectoryPath, "tye.yaml"));
+            var tyeDir = new DirectoryInfo(Path.Combine(tempDirectory.DirectoryPath, ".tye"));
+            using var host = new TyeHost(ConfigFactory.FromFile(projectFile).ToHostingApplication(), Array.Empty<string>())
+            {
+                Sink = sink,
+            };
+
+            await host.StartAsync();
+            try
+            {
+                await WaitUntilSpawned(host.Application);
+
+                var pids = GetAllPids(host.Application);
+                var containers = GetAllContainerIds(host.Application);
+
+                Assert.True(Directory.Exists(tyeDir.FullName));
+                Assert.True(AllRunning(pids));
+                await DockerAssert.AssertAllContainersExistAsync(output, containers);
+
+                await host.PurgeAsync();
+                await Task.Delay(500);
+
+                Assert.False(AnyRunning(pids));
+                await DockerAssert.AssertNonContainersExistAsync(output, containers);
+            }
+            finally
+            {
+                await host.StopAsync();
+            }
+
+            try
+            {
+                tempDirectory.Dispose();
+            }
+            catch (UnauthorizedAccessException) { }
```

Comment:
Usually the cause of this would be things like AV software.

Reply:
lol probably windows defender, since it works on my MacBook